kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/MapColumnReader.java

```diff
@@ -57,10 +59,16 @@ public ColumnVector getDataColumnVector(int batchSize) {
     }
 
     private static Converter[] createElementConverters(
-        int initialBatchSize,
-        MapType typeFromClient,
-        GroupType typeFromFile) {
-        final GroupType innerMapType = (GroupType) typeFromFile.getType("key_value");
+            int initialBatchSize,
+            MapType typeFromClient,
+            GroupType typeFromFile) {
+        // Repeated element can be any name. Latest Parquet versions use "key_value" as the name,
```

Comment:
Can we check which version it was written as?

i.e. latest -> use key_value. older? _only then_ allow arbitrary name

Reply:
As far as I know, there is no clear or confirmed way to identify which version (i.e., Parquet format version) the file was written with. Each writer followed its own way to add the metadata.